crypto/evp/evp_fetch.c

```diff
@@ -85,7 +85,7 @@ static OSSL_METHOD_STORE *get_evp_method_store(OPENSSL_CTX *libctx)
  */
 static uint32_t evp_method_id(unsigned int operation_id, int name_id)
 {
-    if (!ossl_assert(name_id < (1 << 24) || operation_id < (1 << 8))
+    if (!ossl_assert(name_id < (1 << 24) && operation_id < (1 << 8))
```

Comment:
IMO this is better broken into separate assertions so user can see what's wrong.
```
if (!ossl_assert(name_id == 0 || operation_id == 0)
    || !ossl_assert((name_id&0xFFFFFF != name_id)
    || !ossl_assert((operation_id & 0xFF) != operation_id))
       return 0;

Reply:
I think it's a matter of taste how to group these logical conditions and personally I don't think your expression is more intuitive. In view of the above comment, it is natural to see a range check here. In fact, the only improvement I would do is to rearrange the logical conditions to have a separate range assert for each variable:

```
    if (!ossl_assert(name_id > 0 && name_id < (1 << 24))
        || !ossl_assert(operation_id > 0 && operation_id < (1 << 8)))
        return 0;
```

Then in the (unlikely) case of a failure it will be obvious whether `name_id` or `operation_id` was out of range, which should be enough information for the developer.

Reply:
Oh, and it seems like your condition contains a mistake: the first part
```
    !ossl_assert(name_id == 0 || operation_id == 0)
```
inverts the original assertion
```
!ossl_assert(name_id > 0 && operation_id > 0)
```

Reply:
See 55164d68086e534e421ec4ac9753982114664358.